recipes/flex/all/test_package/CMakeLists.txt

```diff
@@ -1,13 +1,6 @@
 cmake_minimum_required(VERSION 3.1)
 project(test_package)
 
-# Find FLEX before `conanbuildinfo.cmake` because that file will let `find_program`
-# look for  executables in host packages (let's hope conan 2.0 fixes this)
```

Comment:
https://cmake.org/cmake/help/latest/module/FindFLEX.html 🤔 Usually it's just variables so it's say https://github.com/conan-io/conan-center-index/pull/13511#discussion_r1008495069

If this was an output of flex's CMake install we could use `build_modules` but I am not sure if thats the case?

Just a note this recipe is missing the cmake target definitions so you'll want to add those

Reply:
You should be able to inject the necessary variables in the CMakeToolchain via `conf`. Generate a small CMake script that sets the necessary Flex variables and add it to `tools.cmake.cmaketoolchain:user_toolchain` so that anything that `tool_requires` Flex will inherit those CMake variables. You probably will only need to set `FLEX_EXECUTABLE`, `FLEX_LIBRARIES`, and `FLEX_INCLUDE_DIRS`. Then maybe see what happens if you rename Conan's generated Config module to something like `_Flex` so that nothing will actually use it.

Reply:
Thanks to @prince-chrismc and @jwillikers for the insights. The suggestion to set up an environment variable gave me an idea. I went and looked at the [code](https://github.com/Kitware/CMake/blob/master/Modules/FindFLEX.cmake) for `FindFLEX`. It's internally using `find_program`, `find_library` and `find_path`, all of which will use `FLEX_ROOT` as the first place to search, for CMake >= 3.12. See also the [doc for `find_library`](http://cmake.org/cmake/help/v3.20/command/find_library.html).

Turns out the following works, at least in my local testing:

* Point `FLEX_ROOT` at the package.
* Keep Conan from replacing CMake's built-in `FindFLEX`. For CMakeDeps, have the recipe produce nothing for CMake, and for the old generators, make the package with a different name.